packages/react-native/ReactCommon/react/renderer/components/progressbar/android/react/renderer/components/progressbar/AndroidProgressBarMeasurementsManager.cpp

```diff
@@ -67,6 +67,7 @@ Size AndroidProgressBarMeasurementsManager::measure(
 
   std::scoped_lock lock(mutex_);
   cachedMeasurement_ = measurement;
+  hasBeenMeasured_ = true;
```

Comment:
The current structure is definitely a bug... But I am wondering if the assumptions around a single global measurement are correct. E.g. would progress bar width change depending on the available width? Same for Switch, if the minimum width is larger than the intrinsic size.

I.e. I think the design of single global measurement itself might be wrong. For text, we use `SimpleThreadSafeCache` to have LRU cache keying against constraints, but at that point, I wonder how much value the cache gives, when we incorporate a bugfix to stop new arch from busting Yoga's own measurement cache.

Reply:
@NickGerleman I think for example Switch always has the same intrinsic size, as far as I understand this is used to provide component's default size. If you take a look at the [onMeasure code of Switch](https://android.googlesource.com/platform/frameworks/support/+/f2e05c341382db64d127118a13451dcaa554b702/appcompat/appcompat/src/main/java/androidx/appcompat/widget/SwitchCompat.java#831), it calculates the size based on some internal assumptions so the width is going to be the same regardless of the available space. 

Looks like the same thing is going on with ProgressBar: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/ProgressBar.java#2196

Reply:
That progress bar example seems like it’s incorporating any available width from the measure spec into set measured width.

I’m also seeing Switch call into superclass measure with the spec, so it seems like it could be reliant on it, but I didn’t test it out (probably simple enough to do with debugger and different styles).

Reply:
I think we’d also give the wrong measurement when using cache for anything with YGMeasureModeExactly.